rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -762,8 +763,11 @@ private <T> List<JRightPadded<T>> visitRightPadded(ASTNode[] nodes, @Nullable St
         }
 
         private Expression insideParentheses(ASTNode node, Function<Space, Expression> parenthesizedTree) {
+            int saveCursor = cursor;
+            whitespace();
             Integer insideParenthesesLevel = getInsideParenthesesLevel(node);
-            if (insideParenthesesLevel != null) {
+            cursor = saveCursor;
+            if (insideParenthesesLevel != null && insideParenthesesLevel > 0) {
```

Comment:
```suggestion
            if (insideParenthesesLevel != null) {
```

I think this is a premature optimisation. Only the Stack object will be created and removed from the heap (a for-loop with zero iterations do not do anything). As a rule of thumb, I think if code can be shorter, most of the time it is more readable. You could also choose to let the `getInsideParenthesesLevel` return a `int`. Maybe that's even better.

Reply:
This is actually a leftover of the mechanism to reset the cursor and can be removed